pkg/cmd/dev/io/os/os.go

```diff
@@ -212,7 +217,25 @@ func (o *OS) CopyFile(src, dst string) error {
 		if err != nil {
 			return err
 		}
-		dstFile, err := os.Create(dst)
+		dstFile, err := os.Open(dst)
```

Comment:
```
dev build short
$ bazel build //pkg/cmd/cockroach-short:cockroach-short //:go_path
WARNING: Ignoring JAVA_HOME, because it must point to a JDK, not a JRE.
Starting local Bazel server and connecting to it...
INFO: Analyzed 2 targets (1601 packages loaded, 23423 targets configured).
INFO: Found 2 targets...
INFO: Elapsed time: 56.219s, Critical Path: 11.48s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
Successfully built binary for target //pkg/cmd/cockroach-short:cockroach-short at cockroach-short
ERROR: open /Users/irfansharif/Software/src/github.com/cockroachdb/cockroach/pkg/ccl/kvccl/kvfollowerreadsccl/boundedstaleness.go: too many open files
```

Existing bug I think, do we want to close these files? Applies to both src and dest.